### ISSUE TITLE
Using python with tasks is impossible

### ISSUE THREAD

**Issue by WeetHet** [zed-industries/zed#8334]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Add a venv to a project, create the following task, write some code in Python that does `input()`, see it fail
```json
[
  {
    "label": "Python main.py",
    "command": "sh",
    "args": ["-c", "python3 main.py"]
  }
]
```

Python code:
```python
a = int(input())
print(a + 3)
```

You should get an error like this:
`invalid literal for int() with base 10: 'source ".../.venv/bin/activate.fish"'`

### Environment

Zed: v0.124.4 (Zed Preview)
OS: macOS 14.2.1
Memory: 8 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

<img width="1440" alt="Screenshot 2024-02-24 at 13 48 39" src="https://github.com/zed-industries/zed/assets/43210583/8c43d3fa-3c59-4a10-becc-7f088ba288c3">

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by SomeoneToIgnore** [comment]
I'm not good with Python, is `Add a venv to a project` important or could things rather be called directly, e.g.
```
{
    "label": "Python main.py (direct)",
    "command": "python3",
    "args": ["main.py"]
}
```
?

**Comment by WeetHet** [comment]
> I'm not good with Python, is `Add a venv to a project` important or could things rather be called directly, e.g.
> 
> ```
> {
>     "label": "Python main.py (direct)",
>     "command": "python3",
>     "args": ["main.py"]
> }
> ```
> 
> ?

Oh, yes, I just forgot to edit it after trying to find a solution

**Comment by WeetHet** [comment]
But having a venv is important as the problem is with it's activation

**Comment by WeetHet** [comment]
This is fixed by #8444 (and successive #9661)
